src/core/event_registry/server_event.rs

```diff
@@ -159,6 +192,13 @@ pub(crate) struct ServerEvent {
     type_id: TypeId,
     type_name: &'static str,
 
+    /// Whether this event depends on replication or not.
+    ///
+    /// Events which don't concern any entity in particular, like a chat message
+    /// event, does not have to wait for replication to be synced. If this field
+    /// is set to `true`, this event will follow this behavior.
```

Comment:
I tweaked it a little bit. Even if event doesn't hold an entity, it still could trigger logic that rely on other entities.

```suggestion
    /// Whether this event depends on replication or not.
    ///
    /// Events like a chat message event, does not have to wait for replication to be synced.
    /// If set to `true`, the event will always be applied immediately.
```

Reply:
Fixed